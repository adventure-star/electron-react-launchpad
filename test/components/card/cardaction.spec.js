import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CardAction from '../../../src/renderer/components/card/cardaction.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(
        <CardAction actions={[{ title: 'link 1', url: 'link url 1' }, { title: 'link 2', url: 'link url 2' }]} />
    );
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Card - Card action component', () => {
    it('should be of type div', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the class "card-action"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card-action');
    });

    it('There should be 2 children', () => {
        const { component } = setupStandard();
        expect(component.children.length).toBe(2);
    });

    it('Both children should be of type <a>', () => {
        const { component } = setupStandard();
        expect(component.children[0].type).toBe('a');
        expect(component.children[1].type).toBe('a');
    });

    it('Both children should have text', () => {
        const { component } = setupStandard();
        expect(component.children[0].children[0]).toBe('link 1');
        expect(component.children[1].children[0]).toBe('link 2');
    });

    it('Both children should have a link', () => {
        const { component } = setupStandard();
        expect(component.children[0].props.href).toBe('link url 1')
        expect(component.children[1].props.href).toBe('link url 2')
    });


    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
