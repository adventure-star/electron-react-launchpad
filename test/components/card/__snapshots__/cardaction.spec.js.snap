// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card - Card action component Matches/Renders a snapshot 1`] = `
<div
  className="card-action"
>
  <a
    href="link url 1"
  >
    link 1
  </a>
  <a
    href="link url 2"
  >
    link 2
  </a>
</div>
`;
